File: binding/android/Bat/bat/src/main/java/ai/picovoice/bat/BatLanguages.java
/*
    Copyright 2026 Picovoice Inc.

    You may not use this file except in compliance with the license. A copy of the license is
    located in the "LICENSE" file accompanying this source.

    Unless required by applicable law or agreed to in writing, software distributed under the
    License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
    express or implied. See the License for the specific language governing permissions and
    limitations under the License.
*/


package ai.picovoice.bat;

/**
 * Bat spoken language understanding engine Languages Object.
 */
public enum BatLanguages {
    UNKNOWN(0),
    DE(1),
    EN(2),
    ES(3),
    FR(4),
    IT(5),
    JA(6),
    KO(7),
    PT(8);

    public static final int numLanguages = 9;

    private final int value;

    private BatLanguages(int value) {
        this.value = value;
    }

    public boolean compare(int i){
        return this.value == i;
    }

    public static BatLanguages fromValue(int _value) {
        BatLanguages[] bls = BatLanguages.values();
        for(int i = 0; i < bls.length; i++) {
            if(bls[i].compare(_value))
                return bls[i];
        }
        return BatLanguages.UNKNOWN;
    }
}

File: binding/ios/BatLanguages.swift
//
//  Copyright 2026 Picovoice Inc.
//  You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
//  file accompanying this source.
//  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
//  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
//  specific language governing permissions and limitations under the License.
//

import PvBat

public enum BatLanguages: Int, CaseIterable {
    case UNKNOWN = 0
    case DE = 1
    case EN = 2
    case ES = 3
    case FR = 4
    case IT = 5
    case JA = 6
    case KO = 7
    case PT = 8
}

extension BatLanguages {
    public static func numLanguages() -> Int {
        return Int(PV_BAT_LANGUAGE_NUM_LANGUAGES)
    }

    public func toString() -> String {
        switch self {
        case .UNKNOWN:
            return "unknown"
        case .DE:
            return "de"
        case .EN:
            return "en"
        case .ES:
            return "es"
        case .FR:
            return "fr"
        case .IT:
            return "it"
        case .JA:
            return "ja"
        case .KO:
            return "ko"
        case .PT:
            return "pt"
        }
    }

    public static func fromString(_ language: String) -> BatLanguages? {
        switch language.lowercased() {
        case "unknown":
            return BatLanguages.UNKNOWN
        case "de":
            return BatLanguages.DE
        case "en":
            return BatLanguages.EN
        case "es":
            return BatLanguages.ES
        case "fr":
            return BatLanguages.FR
        case "it":
            return BatLanguages.IT
        case "ja":
            return BatLanguages.JA
        case "ko":
            return BatLanguages.KO
        case "pt":
            return BatLanguages.PT
        default:
            return nil
        }
    }
}
